src/traces/scattercarpet/attributes.js

```diff
@@ -107,7 +107,8 @@ module.exports = {
         line: extendFlat({},
             {width: scatterMarkerLineAttrs.width},
             colorAttributes('marker'.line)
-        )
+        ),
+        gradient: scatterMarkerAttrs.gradient
```

Comment:
scattercarpet image test in https://github.com/plotly/plotly.js/pull/1620/commits/105ce9b4ffdd77e62bc5bbc06f5e8a5f3f9fdb0a
gl2d fix still to come.